lib/iris/coords.py

```diff
@@ -64,6 +64,23 @@ def name(self, default='unknown'):
         """
         return self.standard_name or self.long_name or self.var_name or default
 
+    def __lt__(self, other):
+        if not isinstance(other, CoordDefn):
+            return NotImplemented
+
+        # Emulate Python 2 behaviour with None
+        key = (self.standard_name is not None, self.standard_name or '',
+               self.long_name is not None, self.long_name or '',
+               self.var_name is not None, self.var_name or '',
+               self.units is not None, self.units or '',
+               self.coord_system is not None, self.coord_system or '')
```

Comment:
Also, thinking about it a bit, those first tests should ensure never comparing `None` with `str`, so maybe I can leave out those `or ''`?